### PR TITLE
fix limit check in search

### LINKED ISSUE #69: Search limit implementation is wrong
[Search method in Client](https://github.com/KnpLabs/packagist-api/blob/master/src/Packagist/Api/Client.php#L64) has a wrong implementation for `limit` (my bad, since it's an implementation of mine)
The `next` response from packagist is an URL, like for example `https://packagist.org/search.json?q=symfony&page=2` and not simply the page itself.
So, the current comparison `response['next'] <= $limit` is meaningless, and always false.

### PR DESCRIPTION
**Pull request by garak**
This will fix #69

### CODE REVIEW

**Comment by robbieaverill** [comment]
Thanks for the patch. Any thoughts on the test failure?

**Comment by robbieaverill** [comment]
Actually nevermind, it's red in the latest master branch build. I'll fix it up soon.

File: src/Packagist/Api/Client.php [hunk 1]
```diff
@@ -76,7 +76,10 @@ public function search(string $query, array $filters = [], int $limit = 0): arra
                 $createResult = [$createResult];
             }
             $results = array_merge($results, $createResult);
-        } while (isset($response['next']) && (0 === $limit || $response['next'] <= $limit));
+            if (isset($response['next'])) {
+                parse_str(parse_url($response['next'], PHP_URL_QUERY), $parse);
+            }
+        } while (isset($response['next']) && (0 === $limit || $parse['page'] <= $limit));
```
Comment (robbieaverill) [inline src/Packagist/Api/Client.php:82]: Do you need to check `isset($parse)`? It looks like it's defined inside the condition abover

Comment (garak) [reply src/Packagist/Api/Client.php:82]: $parse is present along with $response['next'], so logically a check on latter is enough to ensure the existence of former

**Comment by robbieaverill** [comment]
Patch is available in v1.7.1, thanks

**Pull request merged**